src/pages/organize/[orgId]/projects/[campId]/emails/[emailId]/design/index.tsx

```diff
@@ -24,15 +25,32 @@ export const getServerSideProps: GetServerSideProps = scaffold(
   }
 );
 
-const EmailPage: PageWithLayout = () => {
+type Props = {
+  campId: string;
+  emailId: string;
+  orgId: string;
+};
+
+const EmailPage: PageWithLayout<Props> = ({ emailId, orgId }) => {
+  const { data, updateEmail } = useEmail(parseInt(orgId), parseInt(emailId));
   const onServer = useServerSide();
+
   if (onServer) {
     return null;
   }
 
+  if (!data) {
```

Comment:
This is the hackish way that we sometimes do loading states. If the email doesn't exist, we shouldn't even make it to this line (because the page should respond with a 404 before trying to render). But I agree it's not beautiful. We need to take the next step in our refactoring soon.